FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -116,33 +117,66 @@ - (void)resolveShortLink:(NSURL *)url
     @"sdk_version" : FDLSDKVersion
   };
 
-  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSError *error) {
-    NSURL *resolvedURL;
-
-    if (!error && data) {
-      NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-      if ([result isKindOfClass:[NSDictionary class]]) {
-        id invitationIDObject = [result objectForKey:@"invitationId"];
-
-        NSString *invitationIDString;
-        if ([invitationIDObject isKindOfClass:[NSDictionary class]]) {
-          NSDictionary *invitationIDDictionary = invitationIDObject;
-          invitationIDString = invitationIDDictionary[@"id"];
-        } else if ([invitationIDObject isKindOfClass:[NSString class]]) {
-          invitationIDString = invitationIDObject;
+  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSURLResponse *response,
+                                                             NSError *error) {
+    NSURL *resolvedURL = nil;
+    NSHTTPURLResponse *resp = nil;
```

Comment:
nit: maybe rename to something like `completedResponse`

Reply:
As we're primarily interested in `statusCode` I decided to get rid of `resp` object altogether: `NSInteger statusCode = [(NSHTTPURLResponse *)response statusCode];`.